apps/agreement/contracts/disputable/Delay.sol

```diff
@@ -0,0 +1,459 @@
+    function getDelayable(uint256 _id) external view
+        returns (
+            address submitter,
+            uint64 executableAt,
+            DelayableState state,
+            uint256 actionId,
+            bytes script
+        )
+    {
+        Delayable storage delayable = _getDelayable(_id);
+        submitter = delayable.submitter;
+        executableAt = delayable.executableAt;
+        state = delayable.state;
+        actionId = delayable.actionId;
+        script = delayable.script;
+    }
+
+    /**
+    * @dev Tell the information related to the custom token balance submitting permission
+    * @return submitPermissionToken ERC20 token to be used for custom submitting permissions based on token balance
+    * @return submitPermissionBalance Amount of `submitPermissionToken` tokens used for custom submitting permissions
+    * @return challengePermissionToken ERC20 token to be used for custom challenge permissions based on token balance
+    * @return challengePermissionBalance Amount of `challengePermissionToken` tokens used for custom challenge permissions
+    */
+    function getTokenBalancePermission() external view
+        returns (
+            ERC20 submitPermissionToken,
+            uint256 submitPermissionBalance,
+            ERC20 challengePermissionToken,
+            uint256 challengePermissionBalance
+        )
+    {
+        TokenBalancePermission storage submitPermission = submitTokenBalancePermission;
+        submitPermissionToken = submitPermission.token;
+        submitPermissionBalance = submitPermission.balance;
+
+        TokenBalancePermission storage challengePermission = challengeTokenBalancePermission;
+        challengePermissionToken = challengePermission.token;
+        challengePermissionBalance = challengePermission.balance;
+    }
+
+    /**
+    * @dev Tell whether a delayable can be executed or not
+    * @param _id Identification number of the delayable being queried
+    * @return True if the delayable can be executed, false otherwise
+    */
+    function canExecute(uint256 _id) public view returns (bool) {
+        Delayable storage delayable = _getDelayable(_id);
+        return _canExecute(delayable) && _canProceed(delayable.actionId);
+    }
+
+    /**
+    * @dev Tell whether a delayable can be stopped or not
+    * @param _id Identification number of the delayable being queried
+    * @return True if the delayable can be stopped, false otherwise
+    */
+    function canStop(uint256 _id) public view returns (bool) {
+        Delayable storage delayable = _getDelayable(_id);
+        return _canStop(delayable) && _canProceed(delayable.actionId);
+    }
+
+    /**
+    * @notice Schedule a new delayable
+    * @dev IForwarder interface conformance
+    * @param _script Action script to be executed
+    */
+    function forward(bytes _script) public {
+        require(canForward(msg.sender, _script), ERROR_CAN_NOT_FORWARD);
+        _schedule(msg.sender, _script, new bytes(0));
```

Comment:
I think we may want to think about a new forwarder interface that accepts the context bytes, and for now assuming that it is the first forwarder in the path.